src/main/webapp/app/shared/metis/posting-content/posting-content.components.ts

```
@@ -135,46 +137,58 @@ export class PostingContentComponent implements OnInit, OnChanges, OnDestroy {
+                        ),
                     } as Params;
                 } else if (ReferenceType.CHANNEL === referenceType) {
                     // referenceStr: string to be displayed for the reference
-                    referenceStr = this.content.substring(this.content.indexOf(']', patternMatch.startIndex)! + 1, this.content.indexOf('(', patternMatch.startIndex)!);
-                    const channelId = parseInt(this.content.substring(this.content.indexOf('(', patternMatch.startIndex)! + 1, this.content.indexOf(')', patternMatch.startIndex)));
+                    referenceStr = this.content()!.substring(this.content()!.indexOf(']', patternMatch.startIndex)! + 1, this.content()!.indexOf('(', patternMatch.startIndex)!);
+                    const channelId = parseInt(
+                        this.content()!.substring(this.content()!.indexOf('(', patternMatch.startIndex)! + 1, this.content()!.indexOf(')', patternMatch.startIndex)),
+                    );
                     queryParams = {
                         channelId: isNaN(channelId) ? undefined : channelId,
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)